hooks/useDogs.ts

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from "react";
+import { DogDetails } from "../context";
+import { breedsData } from "../data/breedsData";
+import { useLocalStorage } from "./useLocalStorage";
+import { getRandomArrayIndex } from "../utils/getRandomArrayIndex";
+
+type FetchedDog = {
+  message: string;
+  status: string;
+};
+
+type UseDogs = {
+  dogs: DogDetails[];
+  loading: boolean;
+  error?: Error;
+  loadMore: (howMany?: number, setEmpty?: boolean) => Promise<void>;
+};
+
+export const useDogs = (initalHowMany = 10): UseDogs => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error>();
+  const [dogs, setDogs] = useState<DogDetails[]>([]);
+
+  const [storagedBreeds] = useLocalStorage<string[]>(
+    "selectedBreeds",
+    breedsData
+  );
+
+  const loadMore = async (howMany = 4, setEmpty = false) => {
+    if (setEmpty) {
+      return setDogs([]);
+    }
```

Comment:
Doesn't feel right. I'd extract this as another method if needed, it has nothing to do with this one ('load more dogs').